kroxylicious/src/main/java/io/kroxylicious/proxy/config2/api/UnresolvedConfig.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config2.api;
+
+import java.util.List;
+
+public record UnresolvedConfig(List<FilterPlugin> filters,
```

Comment:
How do the virtual cluster definitions fit into this model? I ask as one of the conversations I've been pushing is should filter chains be configured per virtual cluster? 

My instinct is they should, imagine we have two virtual clusters defined which are backed by separate target clusters. 
- VC1 handles PII and thus has the topic encryption and decryption filters configured (and thus wants to de-compress/deserialise & serialise every produce and fetch request) 
- VC2 has no PII and only wants to apply some configuration policy limits (i.e partition counts)

In the current global filter chains model every produce and fetch request passing through VC2 has to be deserialised and serialised (because the topic encryption filter is present) which makes the proxy comparatively expensive to deploy and operate (both in latency and resource consumption terms).

Reply:
This is a fair question, but it's about the config file's structure itself, rather than how the proxy goes about processing the provided config into something it can use. I think we should consider those separately (if nothing else to avoid ending up with a massive PR). My goal with this PR is really to try to replace what we have and figure out the pattern that the runtime should be using for resolving plugins generally, but without delving into the structure of the config file.

Reply:
One thought though: IIUC, currently the virtual cluster abstraction is pluggable. Your request for per-cluster filter chains does make sense, but it suggests we should explore a model where plugins can be nested. We already have other potential use cases for nested plugins. For example, pluggable KMS impls for topic encryption. And even the demo producer/fetch transformation stuff is currently use `Class.forName` where it could using a `ServiceLoader`.

In terms of some of the ideas/concepts of this PR, it suggests that the resolution of plugins (in `Parser.resolvePlugin`) should delegate to a method on `Plugin`, which could then resolve references to other plugin within their own config.

Reply:
Thats fair enough, I just wanted to voice the thought to ensure we aren't closing off that refactoring.